library/aes.c

```diff
@@ -1086,13 +1086,13 @@ int mbedtls_aes_crypt_cbc(mbedtls_aes_context *ctx,
         }
     } else {
         while (length > 0) {
-            mbedtls_xor(output, input, iv, 16);
+            mbedtls_xor(temp, input, iv, 16);
 
-            ret = mbedtls_aes_crypt_ecb(ctx, mode, output, output);
+            ret = mbedtls_aes_crypt_ecb(ctx, mode, temp, iv);
+            memcpy(output, iv, 16);
             if (ret != 0) {
                 goto exit;
             }
-            memcpy(iv, output, 16);
```

Comment:
what's the benefit of moving the memcpy? We're unlikely to hit an error, so don't stall to check `ret` given it's likely to be `0`?

Reply:
I don't know why, but saw a difference in benchmarks.